modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -2,599 +2,753 @@
+    {
+        try {
+            if ($blockonomics_currency == 'btc') {
+                $subdomain = 'www';
+            } else {
+                $subdomain = $blockonomics_currency;
+            }
+
+            $ch = curl_init();
+
+            curl_setopt($ch, CURLOPT_URL, "https://" . $subdomain . ".blockonomics.co/api/price?currency=" . $currency);
+            curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+
+            $contents = curl_exec($ch);
+            if (curl_errno($ch)) {
+                exit('Error:' . curl_error($ch));
+            }
+            curl_close($ch);
+            $price = json_decode($contents)->price;
+            $margin = floatval($this->getMargin());
+            if ($margin > 0) {
+                $price = $price * 100 / (100 + $margin);
+            }
+        } catch (Exception $e) {
+            exit("Error getting price from Blockonomics! {$e->getMessage()}");
+        }
+
+        return intval(1.0e8 * $fiat_amount / $price);
+    }
+
+    /**
+     * If no Blockonomics order table exists, create it
+     */
+    public function createOrderTableIfNotExist()
+    {
+
+        if (!Capsule::schema()->hasTable('blockonomics_bitcoin_orders')) {
+
+            try {
+                Capsule::schema()->create('blockonomics_bitcoin_orders', function ($table) {
+                    $table->increments('id');
+                    $table->integer('id_order');
+                    $table->text('txid');
+                    $table->integer('timestamp');
+                    $table->text('addr');
+                    $table->integer('status');
+                    $table->float('value');
+                    $table->integer('bits');
+                    $table->integer('bits_payed');
+                    $table->string('blockonomics_currency');
+                }
+                );
+            } catch (Exception $e) {
+                exit("Unable to create blockonomics_bitcoin_orders: {$e->getMessage()}");
+            }
+        } else {
+            if (!Capsule::schema()->hasColumn('blockonomics_bitcoin_orders', 'blockonomics_currency')) {
+                Capsule::schema()->table('blockonomics_bitcoin_orders', function ($table) {
+                    $table->string('blockonomics_currency');
+                    // Add btc as the blockonomics_currency for existing orders
+                    try {
+                        Capsule::table('blockonomics_bitcoin_orders')
+                            ->where('blockonomics_currency', '')
+                            ->update([
+                                    'blockonomics_currency' => 'btc'
+                                ]
+                            );
+                    } catch (Exception $e) {
+                        exit("Unable to set default value for existing blockonomics_bitcoin_orders: {$e->getMessage()}");
+                    }
+                });
```

Comment:
This is necessary. There is a simple modifier to set default value https://laravel.com/docs/5.2/migrations#creating-columns